lute/static/js/dict-tabs.js

```diff
@@ -0,0 +1,445 @@
+  const dictTabsLayoutContainer = document.getElementById("dicttabslayout");
+  const iFramesContainer = document.getElementById("dictframes");
+  
+  dictTabsLayoutContainer.style.gridTemplateColumns = `repeat(${columnCount}, minmax(2rem, 8rem))`;
+
+  TABBED_DICTS.forEach((dict, index) => {
+    const domain = getDictURLDomain(dict);
+    const faviconURL = getFavicon(domain);
+    let iFrame = null;
+    let btnLabel = domain.split("www.").splice(-1)[0];
+
+    const btn = createTabBtn(btnLabel, 
+                              dictTabsLayoutContainer, 
+                              index, 
+                              isURLExternal(dict), 
+                              faviconURL);
+
+    if (isURLExternal(dict) == 0) {
+      iFrame = createIFrame(`dict${index}`, iFramesContainer);
+    }
+
+    dictTabButtons.set(btn, iFrame);
+  });
+
+  if (OPTION_DICTS.length > 0) {
+    const selectContainer = document.createElement("div");
+    // const selectList = document.createElement("ol");
+
+    // const selectList = document.createElement("select");
+    const selectButtonBox = document.createElement("div");
+    const domain = getDictURLDomain(OPTION_DICTS[0]);
+    const faviconURL = getFavicon(domain);
+    const btn = createTabBtn(domain.split("www.").splice(-1)[0], 
+                              dictTabsLayoutContainer, 
+                              TERM_DICTS.indexOf(OPTION_DICTS[0]), 
+                              isURLExternal(OPTION_DICTS[0]), 
+                              faviconURL);
+    btn.setAttribute("title", "Right click for dictionary list");
+    const menuImgEl = createImg("", "dict-btn-list-img");
+    btn.appendChild(menuImgEl);
+    const iFrame = createIFrame("selectframe", iFramesContainer);
+    btn.classList.add("dict-btn-select");
+    dictTabButtons.set(btn, iFrame);
+
+    selectContainer.setAttribute("id", "dict-select-container");
+    selectContainer.classList.add("dict-select-list-hide");
+    // selectList.setAttribute("id", "dict-select");
+    selectButtonBox.setAttribute("id", "select-btn-box");
+    selectButtonBox.appendChild(btn); // add select AFTER button
+    selectButtonBox.appendChild(selectContainer); // add select AFTER button
+    // selectContainer.appendChild(selectList); // add select AFTER button
+    dictTabsLayoutContainer.appendChild(selectButtonBox);
+  
+    OPTION_DICTS.forEach((dict) => {
+      const option = document.createElement("p");
+      option.classList.add("dict-select-option");
+      const origIndex = TERM_DICTS.indexOf(dict);
+      const domain = getDictURLDomain(dict);
+      const faviconURL = getFavicon(domain);
+      const faviconEl = createImg(faviconURL, "dict-btn-fav-img");
+      option.textContent = domain.split("www.").splice(-1)[0];
+      option.prepend(faviconEl);
+      option.dataset.dictId = origIndex;
+      option.dataset.dictExternal = isURLExternal(dict);
+      option.dataset.tabOpened = 1 - isURLExternal(dict);
```

Comment:
This line is very hard to understand.  Do you mean it should open a tab, or that a tab is already open?  And isUrlExternal == 1 or 0 only?  Is this even necessary, given that the previous line says "dictExternal = isURLExternal()" ?